File: app/src/main/java/com/yieldzoneandroid/demo/SplashAdActivity.java
package com.yieldzoneandroid.demo;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import androidx.annotation.Nullable;

import com.yieldzone.ads.YieldzoneAdSplashAd;
import com.yieldzone.baselibrary.YieldzoneLoadAdError;
import com.yieldzone.baselibrary.listeners.SplashAdListener;

import static com.yieldzoneandroid.demo.TestPidUtils.YZ_SPLASH_ID;

public class SplashAdActivity extends Activity {

    private Button show_splash;

    private static final String TAG = com.yieldzoneandroid.demo.SplashAdActivity.class.getSimpleName();

    private YieldzoneAdSplashAd mSplash;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_ad_splash);
        show_splash = findViewById(R.id.show_splash);
        show_splash.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showLoadSplash();
            }
        });
    }

    private void showLoadSplash(){

        com.yieldzoneandroid.demo.YZToast.show(com.yieldzoneandroid.demo.SplashAdActivity.this, "splash load...");
        mSplash = new YieldzoneAdSplashAd(com.yieldzoneandroid.demo.SplashAdActivity.this,new SplashAdListener() {
            @Override
            public void onSplashAdDismissed() {
                Log.d(TAG, "onAdClosed()");
                com.yieldzoneandroid.demo.YZToast.show(com.yieldzoneandroid.demo.SplashAdActivity.this, "splash close...");
            }

            @Override
            public void onAdFailedToLoad(int errorCode) {
                Log.d(TAG, "onAdFailedToLoad()");
                com.yieldzoneandroid.demo.YZToast.show(com.yieldzoneandroid.demo.SplashAdActivity.this, "splash error...");
            }

            @Override
            public void onAdFailedToLoad(YieldzoneLoadAdError error) {
                Log.d(TAG, "onAdFailedToLoad()");
                com.yieldzoneandroid.demo.YZToast.show(com.yieldzoneandroid.demo.SplashAdActivity.this, "splash error2...");
            }

            @Override
            public void onSplashAdShowed() {
                Log.d(TAG, "onAdOpened()");
                com.yieldzoneandroid.demo.YZToast.show(com.yieldzoneandroid.demo.SplashAdActivity.this, "splash show...");
            }

            @Override
            public void onSplashAdLoad() {
                Log.d(TAG, "onAdLoaded()");
                mSplash.showAd();
                com.yieldzoneandroid.demo.YZToast.show(com.yieldzoneandroid.demo.SplashAdActivity.this, "splash load...");
            }

            @Override
            public void onSplashAdClicked() {
                Log.d(TAG, "onAdClicked()");
            }

        }, YZ_SPLASH_ID);
        mSplash.loadAd();
    }
}

File: appkt/src/main/java/com/yieldzoneandroid/myapplication/SplashAdActivity.kt
package com.yieldzoneandroid.myapplication

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import com.yieldzone.ads.YieldzoneAdSplashAd
import com.yieldzone.baselibrary.YieldzoneLoadAdError
import com.yieldzone.baselibrary.listeners.SplashAdListener
import com.yieldzoneandroid.myapplication.R

class SplashAdActivity: AppCompatActivity(){
    val TAG: String = "SplashAdActivity"

    var mSplashContainer: ViewGroup? = null

    var mSplash: YieldzoneAdSplashAd? = null

    companion object {
        fun start(context: Context) {
            val intent: Intent = Intent(context, SplashAdActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            context.startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_ad_splash)
        mSplashContainer = findViewById(R.id.splash_container)

        mSplash = YieldzoneAdSplashAd(this@SplashAdActivity,object :
            SplashAdListener {
            override fun onSplashAdDismissed() {
                Log.d(TAG, "onAdClosed()")
                finish()
            }

            override fun onAdFailedToLoad(errorCode: Int) {
                Log.d(TAG, "onAdFailedToLoad()")
            }

            override fun onAdFailedToLoad(error: YieldzoneLoadAdError) {
                Log.d(TAG, "onAdFailedToLoad()")
            }

            override fun onSplashAdShowed() {
              //  TODO("Not yet implemented")
                Log.d(TAG, "onAdOpened()")
            }

            override fun onSplashAdLoad() {
                Log.d(TAG, "onAdLoaded()")
                mSplash?.showAd()
            }

            override fun onSplashAdClicked() {
                Log.d(TAG, "onAdClicked()")
            }

        }, "3DBFEE43-501F-F6DE-3F56-01AE2F44BF40-3")//3DBFEE43-501F-F6DE-3F56-01AE2F44BF40-3
        mSplash!!.loadAd()
    }

}